pkg/querier/http.go

```diff
@@ -326,3 +337,17 @@ func writeError(err error, w http.ResponseWriter) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func (q *Querier) validateEntriesLimits(ctx context.Context, limit uint32) error {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return httpgrpc.Errorf(http.StatusBadRequest, err.Error())
+	}
+
+	maxEntriesLimit := q.limits.MaxEntriesLimitPerQuery(userID)
+	if int(limit) > maxEntriesLimit {
```

Comment:
Same if limit are zero let's no limit anything. This ways our users can get rid of that limits if they want.